### PR TITLE
Read env from mod.rs

### PR DESCRIPTION
**Pull request by nshaaban-cPacket**
Minimize repeated code in all tests

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,4 @@ byte = "0.2.4"
 websocket = "0.26.2"
 thiserror = "1.0"
 native-tls = "0.2.7"
+url = "2.2.2"
```
Comment (1c3t3a) [inline Cargo.toml:27]: Could we skip this dependency by using either `Reqwest`s or `websocket`s URL type? If not that's completely fine, I just wondered.

Comment (nshaaban-cPacket) [reply Cargo.toml:27]: Reqwest and websocket's URL types are a re-export of url's. This also allows us to have socket.rs not reference either Reqwest or Websocket directly (the individual transports should do all the communication with those libraries. 
https://github.com/seanmonstar/reqwest/blob/e6a1a09f0904e06de4ff1317278798c4ed28af66/src/lib.rs#L223
https://github.com/websockets-rs/rust-websocket/blob/b78d08b9bdb401a5ddfb5bea97240af7c49e9795/src/lib.rs#L56

Comment (1c3t3a) [reply Cargo.toml:27]: Ah I see, makes sense!

**Review by 1c3t3a — APPROVED** [review]

**Comment by 1c3t3a** [comment]
I just fixed some errors on the current refactoring branch please rebase **all** open PRs onto the current refactoring branch.

**Comment by 1c3t3a** [comment]
Mho it seems like some of the websocket tests time out after a while. What do you think, is it something that gets fixed within the next commits or do you plan on fixing it now?

**Comment by nshaaban-cPacket** [comment]
> Mho it seems like some of the websocket tests time out after a while. What do you think, is it something that gets fixed within the next commits or do you plan on fixing it now?

Planning on refactoring tests next. Going to use some arcs and atomic bools to make sure the appropriate callbacks actually get called rather than just registered successfully. 

https://github.com/1c3t3a/rust-socketio/pull/61Is also stalled on passing tests.

**Comment by nshaaban-cPacket** [comment]
This PR is priority \#1, without it the other PRs fail tests.

**Pull request merged**